src/config.rs

```diff
@@ -59,19 +79,37 @@ impl From<toml::DecodeError> for Error {
     }
 }
 
+impl Decodable for GpioConfig {
+    fn decode<D: Decoder>(d: &mut D) -> Result<Self, D::Error> {
+        // Get items under the [config] header if present
+        let symlink_root: Option<String> = d.read_struct_field("config", 0, |cfg| {
```

Comment:
Not sure about pulling a function out here, but I can shorten things up a bit by inferring the type `Option<String>` here.